www/src/components/layer-model/layer-content-sections.js

```diff
@@ -0,0 +1,345 @@
+import React from "react"
+import { Link } from "gatsby"
+import MdLoop from "react-icons/lib/md/loop"
+
+import {
+  fontSizes,
+  space,
+  colors,
+  radii,
+  shadows,
+  transition,
+  mediaQueries,
+} from "../../utils/presets"
+
+// Components for building sections used in the model
+const LayerContentWrapper = ({ index, children }) => (
+  <div
+    id={`tabpanel${index}`}
+    aria-labelledby={`tab${index}`}
+    role="tabpanel"
+    css={{
+      padding: `${space[4]} 0`,
+      display: `grid`,
+      gridTemplateRows: `repeat(2, 1fr)`,
+      gridTemplateAreas: `"example" "content"`,
+      gridGap: 0,
+      [mediaQueries.lg]: {
+        gridTemplateRows: `1fr`,
+        gridTemplateAreas: `"example content"`,
+        gridTemplateColumns: `repeat(2, 1fr)`,
+        gridGap: space[6],
+      },
+    }}
+  >
+    {children}
+  </div>
+)
+
+const ExampleWrapper = ({ children }) => (
+  <div
+    css={{
+      borderRadius: radii[3],
+      overflow: `auto`,
+    }}
+  >
+    {children}
+  </div>
+)
+
+// prettier-ignore
+const transitionProperty = `${transition.speed.default} ${transition.curve.default}`
+
+const CodeWrapper = ({
+  title,
+  language,
+  rotateButton = false,
+  sourceIndex,
+  setSourceIndex,
+  children,
+}) => (
+  <React.Fragment>
+    {title && (
+      <div
+        css={{
+          position: `relative`,
+          display: `flex`,
+          alignItems: `center`,
+          justifyContent: `space-between`,
+          borderTopRightRadius: radii[3],
+          borderTopLeftRadius: radii[3],
+        }}
+        className="gatsby-code-title"
+      >
+        <div css={{ fontSize: fontSizes[0] }}>{title}</div>
+        {rotateButton && (
+          <button
+            css={{
+              position: `absolute`,
+              right: space[3],
+              backgroundColor: `transparent`,
+              border: `none`,
+              color: colors.grey[60],
+              cursor: `pointer`,
+              padding: `${space[2]} ${space[2]}`,
+              transition: transitionProperty,
+              borderRadius: radii[2],
+              whiteSpace: `nowrap`,
+              ":focus, :hover, :active": {
+                boxShadow: shadows.floating,
+                color: colors.white,
+              },
+              ":hover": {
+                backgroundColor: colors.purple[40],
+              },
+              ":focus": {
+                backgroundColor: colors.purple[50],
+              },
+              ":active": {
+                backgroundColor: colors.purple[60],
+              },
+              ":focus::before": {
+                content: `"cycle source "`,
+              },
+              ":hover::before": {
+                content: `"cycle source "`,
+              },
+            }}
+            onClick={() => setSourceIndex((sourceIndex + 1) % sources.length)}
```

Comment:
```suggestion
            onClick={() => setSourceIndex((sourceIndex + 1) % sources.length)}
            aria-label="Update code source"
```
The button only has text in it when you focus, which means it will be empty in a screen reader's elements list.